cwl/src/main/scala/cwl/CwlWomExpression.scala

```diff
@@ -49,65 +50,160 @@ final case class InitialWorkDirFileGeneratorExpression(entry: IwdrListingArrayEn
   override def sourceString: String = entry.toString
 
   override def evaluateValue(inputValues: Map[String, WomValue], ioFunctionSet: IoFunctionSet): ErrorOr[WomValue] = {
-    def mustBeString(womValue: WomValue): ErrorOr[String] = womValue match {
-      case WomString(s) => s.validNel
-      case other => WomStringType.coerceRawValue(other).map(_.asInstanceOf[WomString].value).toErrorOr
+    val parameterContext = ParameterContext(inputValues)
+    entry.fold(InitialWorkDirFilePoly).apply(ioFunctionSet, parameterContext, expressionLib)
+  }
+
+  override def evaluateFiles(inputTypes: Map[String, WomValue],
+                             ioFunctionSet: IoFunctionSet,
+                             coerceTo: WomType): ErrorOr[Set[WomFile]] = {
+    "Programmer error: Shouldn't use InitialWorkDirRequirement listing to find output files.".invalidNel
+  }
+
+  /**
+    * We already get all of the task inputs when evaluating, and we don't need to highlight anything else
```

Comment:
this comment seems oddly womtooly

Reply:
Copied from below. Happy to update if there's better text.